api/src/main/java/javax/enterprise/inject/se/SeContainerInitializer.java

```diff
@@ -0,0 +1,317 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package javax.enterprise.inject.se;
+
+import javax.enterprise.inject.spi.AnnotatedType;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.Extension;
+import java.lang.annotation.Annotation;
+import java.util.Map;
+import java.util.ServiceConfigurationError;
+import java.util.ServiceLoader;
+
+/**
+ * Container initializer to bootstrap CDI in Java SE.
+ * It is obtained by calling the {@link SeContainerInitializer#getInstance()} static method
+ * <p>
+ * <p>
+ * Typical usage looks like this:
+ * </p>
+ * <p>
+ * <pre>
+ * SeContainer<Object> container = SeContainerInitializer.getInstance().initialize();
+ * container.select(Foo.class).get();
+ * container.event().select(Bar.class).fire(new Bar());
+ * container.close();
+ * </pre>
+ * <p>
+ * <p>
+ * The {@link SeContainer} interface implements AutoCloseable:
+ * </p>
+ * <p>
+ * <pre>
+ * try (SeContainer<Object> container = SeContainerInitializer.getInstance().initialize()) {
+ *     container.select(Foo.class).get();
+ * }
+ * </pre>
+ * <p>
+ * <p>
+ * By default, the discovery is enabled so that all beans from all discovered bean archives are considered. However, it's possible to define a "synthetic" bean
+ * archive, or the set of bean classes and enablement respectively:
+ * </p>
+ * <p>
+ * <pre>
+ * SeContainer<Object> container = SeContainerInitializer.getInstance().beanClasses(Foo.class, Bar.class).alternatives(Bar.class).initialize());
+ * </pre>
+ * <p>
+ * <p>
+ * Moreover, it's also possible to disable the discovery completely so that only the "synthetic" bean archive is considered:
+ * </p>
+ * <p>
+ * <pre>
+ * SeContainer<Object> container = SeContainerInitializer.getInstance().disableDiscovery().beanClasses(Foo.class, Bar.class).initialize());
+ * </pre>
+ * <p>
+ * <p>
+ * <p>
+ * In the same manner, it is possible to explicitly declare interceptors, decorators, extensions and implementation specific options using the builder.
+ * </p>
+ * <p>
+ * <pre>
+ * SeContainerInitializer containerInitializer = SeContainerInitializer.getInstance()
+ *    .disableDiscovery()
+ *    .addPackages(Main.class, Utils.class)
+ *    .addInterceptors(TransactionalInterceptor.class)
+ *    .addProperty("property", true);
+ * SeContainer<Object> container = container.initialize();
+ * </pre>
+ *
+ * @author Antoine Sabot-Durand
+ * @author Martin Kouba
+ * @author John D. Ament
+ * @since 2.0
+ */
+public abstract class SeContainerInitializer {
+
+    protected static volatile SeContainerInitializer seContainerInitializer = null;
+
+
+    /**
+     * Returns an instance of {@link SeContainerInitializer}
+     * Each call returns a new instance
+     *
+     * @return a new SeContainerInitializer instance.
+     * @throws IllegalStateException if called in a Java EE container
+     */
+    public static SeContainerInitializer getInstance() {
```

Comment:
It may be assumed but the impl is not correct. Right now, it will always return the cached initializer. Also it might be useful to throw an exception if multiple `SeContainerInitializer` impls are found.